examples/server/server.cpp

```diff
@@ -2768,7 +2768,13 @@ static void append_to_generated_text_from_generated_token_probs(llama_server_con
 }
 
 std::function<void(int)> shutdown_handler;
-inline void signal_handler(int signal) { shutdown_handler(signal); }
+std::atomic_flag is_terminating = ATOMIC_FLAG_INIT;
+inline void signal_handler(int signal) {
+    if (is_terminating.test_and_set()) {
+        exit(0); // force exit the process, in case it hangs
```

Comment:
Since this is an unclean exit, wouldn't it be better to `exit(1)`?

Reply:
Yeah that's reasonable, I'll do that if we all agree that it's a good idea to add this such feature to the code base.

I actually mentioned in the description that this is a sketchy solution, so I'm not even sure if it's a good idea or not. Just a small stuff to have a better developer experience.

Reply:
I added the message and exit(1), also I put a comment to say that this is a "hotfix" that can be removed when the server become more stable